x-pack/test/observability_functional/apps/observability/pages/overview/alert_table.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../../ftr_provider_context';
+
+const ALL_ALERTS = 10;
+
+export default ({ getPageObjects, getService }: FtrProviderContext) => {
+  const PageObjects = getPageObjects(['header']);
+  const esArchiver = getService('esArchiver');
+
+  describe('Observability overview', function () {
+    this.tags('includeFirefox');
+
+    const observability = getService('observability');
+    const retry = getService('retry');
+
+    before(async () => {
+      await esArchiver.load('x-pack/test/functional/es_archives/observability/alerts');
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/observability/alerts');
+    });
+
+    describe('Without alerts', function () {
+      it('navigate and open alerts section', async () => {
+        await observability.overview.common.navigateToOverviewPage();
+        await PageObjects.header.waitUntilLoadingHasFinished();
+        await observability.overview.common.openAlertsSectionAndWaitToAppear();
+      });
+
+      it('should show no data message', async () => {
+        await retry.try(async () => {
+          await observability.overview.common.getAlertsTableNoDataOrFail();
+        });
+      });
+    });
```

Comment:
If I understand correctly, the first test case navigates to the overview page and asserts the alerts section opens, while the second test case asserts on no data in the alerts table. 

But the second test case does not seem to navigate to the overview page. 

I might not understand totally how the functional tests are setup, but I think tests should not depend on a state set by another test. 
I think the order of execution is not guaranteed, and therefore the 2nd test might run before the first one. And even if the order of execution is guaranteed, if someone adds a new test in between, it might make the next test fail. 

I suggest to write the tests as they were alone with all their necessary setup, e.g. navigating to the overview page, waiting for the headers, etc...
In this case we could also just merge the two test cases into one? What do you think?

Also, this might explain why you had some flakiness earlier.

Reply:
Hey Kevin,
Thanks for your comment. Indeed tests are related to each other and we cannot change order or insert a step in between without ensuring everything will work fine in the new setup.

This is the expected behavior in functional tests, functional tests are like a narrative of the user's interaction with our system and the test is set up in a way that it will not continue without first making sure previous steps are successful. As a result of this setup, we cannot run one test case in isolation and we need to run the whole test suite.

The flakiness is related to `await observability.overview.common.openAlertsSectionAndWaitToAppear();` and I will test another idea to make sure it will not cause any issue in future :)